# Viral Ancestral
Infer a phylogenetic tree from viral sequences, and use the tree to estimate the ancestral protein sequence of this group.

## To do 
- Figure out how to increase substitution cost for Replicase gene and run through MACSE
- Run Replicase gene with higher substitution cost through trimAL
- Download T-COFFEE for Mac
- Run Replicase Gene through T-COFFEE
- Manually align YX-AUG regions separately (IRES types 1,2,3) with Helena

## Data before alignment: 


## Variations on Alignments: 
- Replicase NT through MACSE with default parameters, then trimAL, IRES type 1: 
- Replicase AA through MACSE with default parameters, then trimAL IRES type 1:
- Replicase NT through MACSE with default parameters, then trimAL, IRES type 2: 
- Replicase AA through MACSE with default parameters, then trimAL IRES type 2:
- Replicase NT through MACSE with default parameters, then trimAL, IRES type 3: 
- Replicase AA through MACSE with default parameters, then trimAL IRES type 3:
- 5UTR NT through MACSE with default parameters, then trimAL, IRES type 1: 
- 5UTR AA through MACSE with default parameters, then trimAL IRES type 1:
- 5UTR NT through MACSE with default parameters, then trimAL, IRES type 2: 
- 5UTR AA through MACSE with default parameters, then trimAL IRES type 2:
- 5UTR NT through MACSE with default parameters, then trimAL, IRES type 3: 
- 5UTR AA through MACSE with default parameters, then trimAL IRES type 3:
